opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -1441,7 +1441,7 @@ namespace Opm
         double relaxation_factor = 1.;
         const double min_relaxation_factor = 0.6;
         bool converged = false;
-        int stagnate_count = 0;
+        [[maybe_unused]] int stagnate_count = 0;
```

Comment:
I guess clang is smart enough to realise that nothing reads the [incremented count](https://github.com/OPM/opm-simulators/blob/900056e8d719d1569b1d6d6e4cbd7d624f2f95b6/opm/simulators/wells/MultisegmentWell_impl.hpp#L1523) once it's complete.  In any case, there's more work coming in this area&ndash;notably a refactoring to extract the common parts of `iterateWellEqWithSwitching()` and `iterateWellEqWithControl()`, so this is a good fix in the short term.

Reply:
yeah, clang is getting annoyingly good at looking through if (false) shenanigans etc ;)